sdk/src/cose_validator.rs

```diff
@@ -814,6 +828,76 @@ fn get_timestamp_info(sign1: &coset::CoseSign1, data: &[u8]) -> Result<TstInfo>
     Err(Error::NotFound)
 }
 
+#[async_generic(async_signature( th: &dyn TrustHandlerConfig, chain_der: &[Vec<u8>], cert_der: &[u8], validation_log: &mut impl StatusTracker))]
+#[allow(unused)]
+fn check_trust(
+    th: &dyn TrustHandlerConfig,
+    chain_der: &[Vec<u8>],
+    cert_der: &[u8],
+    validation_log: &mut impl StatusTracker,
+) -> Result<bool> {
+    // is the certificate trusted
+
+    let verify_result: Result<bool> = if _sync {
+        #[cfg(not(feature = "openssl_sign"))]
+        {
+            Err(Error::NotImplemented(
+                "no trust handler for this feature".to_string(),
+            ))
+        }
+
+        #[cfg(feature = "openssl_sign")]
+        {
+            verify_trust(th, chain_der, cert_der)
+        }
+    } else {
+        #[cfg(target_arch = "wasm32")]
+        {
+            verify_trust_async(th, chain_der, cert_der).await
+        }
+
+        #[cfg(feature = "openssl_sign")]
+        {
+            verify_trust(th, chain_der, cert_der)
+        }
+
+        #[cfg(all(not(feature = "openssl_sign"), not(target_arch = "wasm32")))]
+        {
+            Err(Error::NotImplemented(
+                "no trust handler for this feature".to_string(),
+            ))
+        }
+    };
+
+    match verify_result {
+        Ok(trusted) => {
+            if trusted {
+                let log_item =
+                    log_item!("Cose_Sign1", "signing certificate trusted", "verify_cose")
+                        .validation_status(validation_status::SIGNING_CREDENTIAL_TRUSTED);
+                validation_log.log_silent(log_item);
+                Ok(true)
+            } else {
+                let log_item =
+                    log_item!("Cose_Sign1", "signing certificate untrusted", "verify_cose")
+                        .error(Error::CoseCertUntrusted)
```

Comment:
It seems weird having errors we don't actually ever return as errors, but only in validation. At some point we may want to put these in a separate category.

Reply:
I will force this one to return upon further reflection.